quotes.py

```diff
@@ -0,0 +1,258 @@
+            resolved = await self.client(GetFullChannelRequest(reply.fwd_from.channel_id))
+            chat = resolved.chats[0]
+            user_id = chat.id
+            fwd_status = 'channel_open'  # now it's just regular channel, only saved_from_peer was the difference.
+
+        if fwd_status not in ['hidden', 'channel_private']:
+            if not await self.client.get_profile_photos(user_id):
+                no_pfp = True
+            else:
+                no_pfp = False
+        else:
+            no_pfp = True
+
+        if fwd_status not in ['hidden', 'channel_private']:
+            pfp_bytes = BytesIO()
+            await self.client.download_profile_photo(user_id, file=pfp_bytes)
+            pfp_bytes.seek(0)
+            if no_pfp is False:
+                pfp = 'data:image/png;base64, ' + base64.b64encode(pfp_bytes.read()).decode()
+            else:
+                pfp = ''
+            pfp_bytes.close()
+
+        if fwd_status not in ['hidden', 'channel_private']:
+            colors = ["#fb6169", "#85de85", "#f3bc5c", "#65bdf3", "#b48bf2", "#ff5694", "#62d4e3", "#faa357"]
+            num1 = user_id % 7
+            num2 = [0, 7, 4, 1, 6, 3, 5]
+            colour = colors[num2[num1]]
+        else:
+            colour = '#b48bf2'
+
+        if fwd_status == 'no_fwd' and isinstance(message.chat, Channel):
+            participant = await self.client(GetParticipantRequest(message.chat_id, user_id))
+            user = participant.users[0]
+        elif fwd_status not in ['hidden', 'channel_open', 'channel_private']:
+            user = (await self.client(GetFullUserRequest(user_id))).user
+
+        if fwd_status == 'channel_open':
+            username = chat.title
+        elif fwd_status == 'hidden':
+            username = reply.fwd_from.from_name
+        elif fwd_status == 'channel_private':
+            username = 'Hidden.'
+        else:
+            username = user.first_name
+            if user.last_name:
+                username += f' {user.last_name}'
+
+        if fwd_status.startswith('channel'):
+            admintitle = 'channel'
+        elif fwd_status == 'hidden':
+            admintitle = ''
+        else:
+            try:
+                admintitle = participant.participant.rank
+                if not admintitle:
+                    admintitle = 'admin'
+            except (AttributeError, UnboundLocalError):
+                admintitle = ''
+
+        data = ujson.dumps({
+            "ProfilePhotoURL": pfp,
+            "UsernameColor": colour,
+            "Username": username,
+            "AdminTitle": admintitle,
+            "Text": reply.raw_text,
+            "Markdown": get_markdown(reply),
+            "Template": args[0].lower(),
+            "APIKey": self.config["API_TOKEN"]
+        })
+        requested = requests.post(f'{self.config["API_URL"]}/api/v2/quote', data=data)
+
+        requested = requested.json()
+
+        if requested['status'] == 500:
```

Comment:
No. If an error happens, it should be an error. Use .raise_for_status(), assuming the server sends the http status the same as the JSON status

Reply:
no it doesn't, it sends 200 in all conditions

Reply:
Well that's just stupid